### PR TITLE
AM-1129:  Pact search query Consumer Test

### PR DESCRIPTION
**Pull request by SivaKanukolluQA**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-1129

### Change description ###
PACT Implementation for POST /am/role-assignments/query

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTestForSearchQuery.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    private DslPart createRoleAssignmentResponseSearchQuery() {
+        return newJsonBody(o -> o
+            .minArrayLike("links", 1, 1, link -> link.stringType("rel", "binary")
+                    .stringType("href", "http://localhost:4096/am/role-assignments/query"))
+            .minArrayLike("roleAssignmentResponse", 1, 1,
+                    roleAssignmentResponse -> roleAssignmentResponse
+                            .stringType("id", "14a21569-eb80-4681-b62c-6ae2ed069e6f")
+                            .stringValue("actorIdType", "IDAM")
+                            .stringValue("actorId", ACTOR_ID)
```
Comment (aashishaggarwal123) [inline src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTestForSearchQuery.java:111]: matching the actor id can cause issue. It should be based upon type but check with provider.

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTestForSearchQuery.java [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+
+        JSONObject jsonResponse = new JSONObject(actualResponseBody);
+        JSONArray roleAssignmentResponse = (JSONArray)jsonResponse.get("roleAssignmentResponse");
+        JSONObject first = (JSONObject)roleAssignmentResponse.get(0);
+        JSONObject attributes = (JSONObject) first.get("attributes");
+        assertThat(attributes.get("primaryLocation"), equalTo("500A2S"));
+        assertThat(attributes.get("jurisdiction"), equalTo("IA"));
+    }
+
+    @Test  //Need to fix assertion for second record.
```
Comment (ConorH994) [inline src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTestForSearchQuery.java:244]: remove comment if resolved

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTestForSearchQuery.java [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+                        .then()
+                        .log().all().extract().asString();
+
+        JSONObject jsonResponse = new JSONObject(actualResponseBody);
+        JSONArray roleAssignmentResponse = (JSONArray)jsonResponse.get("roleAssignmentResponse");
+
+        JSONObject first = (JSONObject)roleAssignmentResponse.get(0);
+        assertThat(first.get("actorId"), equalTo("ca93ea54-c219-4c6d-add6-e687a0f1f5f7"));
+        /*JSONObject second = (JSONObject)roleAssignmentResponse.get(1);
+        assertThat(second.get("actorId"), equalTo("fa4c86ba-289c-4227-b924-13f55929047c"));*/
```
Comment (ConorH994) [inline src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTestForSearchQuery.java:264]: Remove if not required

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTestForSearchQuery.java [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+                                .stringValue("roleType", "ORGANISATION")
+                                .stringValue("roleName", "tribunal-caseworker")
+                                .stringValue("classification", "PUBLIC")
+                                .stringValue("grantType", "STANDARD")
+                                .stringValue("roleCategory", "STAFF")
+                                .booleanValue("readOnly", false)
+                                .object("attributes", attribute -> attribute
+                                        .stringType("jurisdiction", "IA")
+                                        .stringType("primaryLocation", "219ASA"))
+                )*/
```
Comment (ConorH994) [inline src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTestForSearchQuery.java:338]: Sonar will complain about commented code

**Review by aashishaggarwal123 — APPROVED** [review]

**Pull request merged**